mobile/library/common/network/apple_proxy_resolution.cc

```diff
@@ -0,0 +1,27 @@
+#include "library/common/network/apple_proxy_resolution.h"
+
+#include <memory>
+
+#include "library/common/api/external.h"
+#include "library/common/network/apple_proxy_resolver.h"
+#include "library/common/network/proxy_types.h"
+
+// NOLINT(namespace-envoy)
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+void register_apple_proxy_resolver() {
+  auto resolver = std::make_unique<Envoy::Network::AppleProxyResolver>();
+  resolver->start();
+
+  auto api = std::make_unique<Envoy::Network::ProxyResolverApi>();
+  api->resolver = std::move(resolver);
+
+  Envoy::Api::External::registerApi("envoy_proxy_resolver", api.release());
```

Comment:
Why do we `release` the `api` ownership here? Should this be `api.get()` instead?

Reply:
Ah, I didn't know `Envoy::Api::External::registerApi` takes the ownership. Does it wrap it in some kind of smart pointer inside it?